docs/presentation/slides.md

```diff
@@ -0,0 +1,507 @@
+
+1. Steer kite to centre of wind window.
+2. Wind pulls the kite, unwinding the string, generating power.
+3. Steer kite to edge of wind window.
+4. Wind string back in, using less energy than was generated on the way out.
+5. Repeat.
+
+---
+
+# Hardware
+
+What hardware do we need?
+
+1. Two motors
+2. Spools to wind the strings on
+3. Motor controllers
+4. A microcontroller and other electronics to manage the motors
+5. A big battery to power it all
+
+Let's take the tires off a hoverboard!
+
+---
+
+# Hoverboard
+
+David bought a "Zinc Smart GT Pro" hoverboard.
+
+These boards have been hacked to make
+[DIY mobility devices](https://hackaday.io/project/170932-hoverboards-for-assistive-devices). Their
+[firmware](https://github.com/gearsincorg/Hoverboards-for-assistive-devices) is all in C, but served
+as useful documentation and inspiration. They also have some
+[circuit diagrams and pinouts](https://github.com/gearsincorg/Hoverboards-for-assistive-devices/blob/master/Hoverboard/Documents/Schematic_CoolAndFun.pdf)
+which were very useful.
+
+---
+
+![The front of the board](board_front.jpg)
+
+---
+
+![The back of the board](board_back.jpg)
+
+---
+
+# Hoverboard boards
+
+Each half has an almost-identical "TT-SD2.2" board with:
+
+- A GD32F130C8 microcontroller.
+- 6 power MOSFETs to drive the three phase brushless DC motor, and 3 hall effect sensors to sense
+  its position.
+- A BMI160 6-axis IMU (accelerometer & gyroscope).
+- Opto-interruptors to sense someone standing on the hoverboard.
+- Two UARTs.
+- Some LEDs.
+- Battery voltage measurement.
+- Motor current measurement.
+- A power button.
+- An (unpopulated) SWD header.
+
+One half also has a piezo speaker, a battery charger connection, and a Bluetooth module. The only
+connections between the two halves are power, and a serial connection between the UARTs for
+communication.
+
+---
+
+# Hardware hacking
+
+We removed the tires from the hoverboard wheels so they could serve as spools, and soldered headers
+onto the SWD port and second serial port.
+
+Connect an ST-Link V2 SWD debugger, and we're in! First step, disable the watchdog timer:
+
+```sh
+openocd -f interface/stlink-v2.cfg -f target/stm32f1x.cfg -c init \
+  -c "reset halt; stm32f1x options_write 0 SWWDG"
+```
+
+---
+
+![The board connected to an STLink debugger](board_stlink.jpg)
+
+A USB serial adapter is handy too.
+
+???
+
+We stuck some tape on the piezo speaker as it was too loud.
+
+---
+
+# Time to write some firmware
+
+It would be easiest to write it in C with PlatformIO and the official GigaDevice headers, based on
+the existing assistive devices code we found...
+
+--
+
+Let's not do that.
+
+---
+
+# Embedded Rust
+
+- Peripheral Access Crate: mostly-safe access to memory-mapped peripheral registers, autogenerated
+  from CMSIS-SVD files by `svd2rust`.
+- HAL crate: higher level wrappers around various peripherals, generally implementing traits from
+  `embedded-hal`.
```

Comment:
* diagram of how the PAC and HAL stuff fits together (layer diagram with hardware at the bottom or something, and application code at the top?)
* same diagram with "stm32-rs?" stamped on each of the bits that they have a version of (and `A GD32F130 is almost the same as an STM32F130, right?` at the bottom of the slide, followed by `Nope`)

the rest goes in speaker notes

Reply:
I think I prefer it this way.